components/sp-seamless-provision/src/main/java/com/wso2telco/sp/discovery/LocalDiscovery.java

```diff
@@ -85,21 +87,31 @@ private ServiceProviderDto checkRemotlyDiscovery(DiscoveryServiceConfig discover
         return serviceProviderDto;
     }
 
-    private boolean checkLocally(boolean isPcrServiceEnabled, String sectorId, String clientId, String clientSecret)
-            throws DicoveryException {
+    private boolean checkLocally(boolean isCacheOverride, boolean isPcrServiceEnabled, String sectorId, String clientId,
+                                 String clientSecret) throws DicoveryException {
         boolean isAppAvailable = false;
         if (isPcrServiceEnabled) {
             isAppAvailable = checkSpAvailabilityInmemory(sectorId, clientId);
+            if (isCacheOverride && !isAppAvailable) {
+                isAppAvailable = dataStoreValidation(clientId, clientSecret);
+            }
         } else {
 
-            AdminServiceDto adminServiceDto = provisioningService.getOauthServiceProviderData(clientId,this.spProvisionDto);
+            isAppAvailable = dataStoreValidation(clientId, clientSecret);
+        }
+        return isAppAvailable;
+    }
 
-            if (clientSecret != null && !clientSecret.isEmpty() && adminServiceDto != null
-                    && adminServiceDto.getOauthConsumerSecret() != null
-                    && !adminServiceDto.getOauthConsumerSecret().isEmpty()
-                    && adminServiceDto.getOauthConsumerSecret().equals(clientSecret)) {
-                isAppAvailable = true;
-            }
+    private boolean dataStoreValidation(String clientId, String clientSecret) {
+        boolean isAppAvailable = false;
+        AdminServiceDto adminServiceDto = provisioningService.getOauthServiceProviderData(clientId,
+                this.spProvisionDto);
+
+        if (clientSecret != null && !clientSecret.isEmpty() && adminServiceDto != null
```

Comment:
instead of 
clientSecret != null && !clientSecret.isEmpty()
better to use
StringUtils.isNotEmpty(clientSecret)